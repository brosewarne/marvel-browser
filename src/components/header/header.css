.appHeader {
    padding-right: 10px;
    color: white;
    text-align: center;
}

.appTitle {
    font-family: "AmericanCaptain";
    font-weight: normal;
    font-size: 3rem;
    text-align: center;
}
