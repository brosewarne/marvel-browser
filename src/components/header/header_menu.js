import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from 'material-ui/Button';
import Menu, { MenuItem } from 'material-ui/Menu';
import Typography from 'material-ui/Typography';

class HeaderMenu extends React.Component {
    static propTypes = {
        menuName: PropTypes.string.isRequired,
        menuOptions: PropTypes.array.isRequired
    };

    constructor(props) {
        super(props);

        this.state = {
            anchorEl: null
        };

        this.menuItems = this.getMenuItems();
    }

    /**
     * Open the menu when the menu button is clicked
     * @param {Object} event - The click event
     */
    onClick = (event) => {
        this.setState({ anchorEl: event.currentTarget });
    };

    /**
     * Close the menu when an option is clicked
     */
    onMenuOptionClick = () => {
        this.setState({ anchorEl: null });
    };

    /**
     * Create MenuItem components from this.props.menuOptions
     * @returns {Array} - The array of MenuItem components
     */
    getMenuItems = () => {
        return this.props.menuOptions.map(
            (menuOption) => {
                return (
                    <MenuItem
                        onClick={this.onMenuOptionClick}
                    >
                        <Link
                            href={menuOption.href}
                            to={{ pathname: menuOption.href }}
                        >
                            <Typography variant="subheading" color="primary" align="center">
                                {menuOption.name}
                            </Typography>
                        </Link>
                    </MenuItem>
                );
            }
        );
    };

    render() {
        const { anchorEl } = this.state;

        return (
            <Fragment>
                <Button
                    aria-owns={anchorEl ? 'simple-menu' : null}
                    aria-haspopup="true"
                    onClick={this.onClick}
                    color="secondary"
                    variant="raised"
                >
                    {this.props.menuName}
                </Button>
                <Menu
                    id="simple-menu"
                    anchorEl={anchorEl}
                    open={Boolean(anchorEl)}
                    onClose={this.handleClose}
                >
                    {this.menuItems}
                </Menu>
            </Fragment>
        );
    }
}

export { HeaderMenu };
