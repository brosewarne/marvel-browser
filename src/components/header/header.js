import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';

import { HeaderMenu } from './header_menu';

import './header.css';

class Header extends Component {
    constructor(props) {
        super(props);

        this.menuOptions = ['characters', 'comics', 'series'].reduce(
            (allOptions, itemType) => {
                // eslint-disable-next-line no-param-reassign
                allOptions[itemType] = [
                    {
                        name: 'Search'
                    },
                    {
                        name: 'Browse',
                        href: `/${itemType}/A/1`
                    },
                    {
                        name: 'Random',
                        href: `/${itemType}/random`
                    }
                ];
                return allOptions;
            }, {});
    }

    render = () => {
        return (
            <AppBar position="static">
                <Toolbar>
                    <Typography color="primary" className="appHeader" align="center">
                        <Link to="/" href="/">
                            <h1 className="appTitle">Marvel Browser</h1>
                        </Link>
                    </Typography>
                    <HeaderMenu menuName="Characters" menuOptions={this.menuOptions.characters} />
                    <HeaderMenu menuName="Comics" menuOptions={this.menuOptions.comics} />
                    <HeaderMenu menuName="Series" menuOptions={this.menuOptions.series} />
                </Toolbar>
            </AppBar>
        );
    };
}

export { Header };
