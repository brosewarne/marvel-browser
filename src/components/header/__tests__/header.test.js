import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { Header } from '../header';

configure({ adapter: new Adapter() });

describe('<Header />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <Header />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
