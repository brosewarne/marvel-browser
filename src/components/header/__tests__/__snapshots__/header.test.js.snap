// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      align="center"
      className="appHeader"
      color="primary"
    >
      <Link
        href="/"
        replace={false}
        to="/"
      >
        <h1
          className="appTitle"
        >
          Marvel Browser
        </h1>
      </Link>
    </WithStyles(Typography)>
    <HeaderMenu
      menuName="Characters"
      menuOptions={
        Array [
          Object {
            "name": "Search",
          },
          Object {
            "href": "/characters/A/1",
            "name": "Browse",
          },
          Object {
            "href": "/characters/random",
            "name": "Random",
          },
        ]
      }
    />
    <HeaderMenu
      menuName="Comics"
      menuOptions={
        Array [
          Object {
            "name": "Search",
          },
          Object {
            "href": "/comics/A/1",
            "name": "Browse",
          },
          Object {
            "href": "/comics/random",
            "name": "Random",
          },
        ]
      }
    />
    <HeaderMenu
      menuName="Series"
      menuOptions={
        Array [
          Object {
            "name": "Search",
          },
          Object {
            "href": "/series/A/1",
            "name": "Browse",
          },
          Object {
            "href": "/series/random",
            "name": "Random",
          },
        ]
      }
    />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
